apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableVersion.cs

```diff
@@ -0,0 +1,199 @@
+    /// is based on the current timestamp micros. For those columns, other reads and writes should
+    /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+    /// a custom versioning scheme with 64-bit values.
+    /// </para>
+    /// </remarks>
+    public struct BigtableVersion : IComparable, IComparable<BigtableVersion>, IEquatable<BigtableVersion>
+    {
+        private const long TicksPerMicro = 10;
+
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersion"/> value from a 64-bit value.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note: when using ReadModifyWriteRow, modified columns automatically use a server version, which
+        /// is based on the current timestamp micros. For those columns, other reads and writes should
+        /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+        /// a custom versioning scheme with 64-bit values.
+        /// </para>
+        /// </remarks>
+        /// <param name="value">
+        /// The non-negative version value, or -1 to initialize from the microseconds of DateTime.UtcNow.
+        /// </param>
+        public BigtableVersion(long value)
+        {
+            GaxPreconditions.CheckArgumentRange(value, nameof(value), -1, long.MaxValue);
+            Value = value == -1 ? DateTime.UtcNow.Ticks / TicksPerMicro : value;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersion"/> value from the microseconds of a timestamp.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note: when using ReadModifyWriteRow, modified columns automatically use a server version, which
+        /// is based on the current timestamp micros. For those columns, other reads and writes should
+        /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+        /// a custom versioning scheme with 64-bit values.
+        /// </para>
+        /// </remarks>
+        /// <param name="timestamp">
+        /// The timestamp whose microseconds should be used as the version value. It must be specified in UTC.
+        /// </param>
+        public BigtableVersion(DateTime timestamp)
+        {
+            GaxPreconditions.CheckArgument(
+                timestamp.Kind == DateTimeKind.Utc,
+                nameof(timestamp),
+                $"The {nameof(BigtableVersion)} timestamp must be specified in UTC.");
+
+            Value = timestamp.Ticks / TicksPerMicro;
+        }
+
+        /// <summary>
+        /// Gets the version value. Greater version values indicate newer cell values.
+        /// </summary>
+        public long Value { get; }
+
+        /// <summary>
+        /// Gets the DateTime equivalent to the version assuming the value is a timestamp micros value.
```

Comment:
I see we're using the .NET epoch (0001-01-01T00:00:00). Is that what other languages are likely to do? Might the Unix epoch work better? (Just thinking about interop really.)

Reply:
Oops, yes you're right. I'll fix this (in code and docs)